### PR TITLE
Jetpack Focus: Enhance screenshots

### PR DESCRIPTION
**Pull request by zwarm**
Parent #16900

This PR includes:

- New methods added to `JPScreenshotTest` to generate the following screenshots
-- Edit blog post
-- Notifications
-- Media
-- Blogging Reminders
-- Blogging Prompts
-- Choose site topic (launched via Site Creation)
-- Create new options (post, story, page) - bottom sheet shown after FAB click
-- Choose a layout (after clicking new page)
- A `Screenshots` enum was introduced to manage which screenshots are enabled for generation. Before generation a check is made to validate if the screenshot is enabled. 
- Includes method name refactoring
- Includes reorganization of methods within the class
- Adds a few helpers to support running the screenshots in any order
- Fixes the `EditBlog` screenshot within WPScreenshotTest mentioned here #16846 
- Add mocks for blogging prompts, layouts and site topics

**Notes**
Not all screenshots mentioned above are enabled for generation, but the logic exists to generate them (minus some content in the mocks). 

To test:
1. Checkout the PR branch locally 
2. Switch to `jetpackJalapenoDebug` build variant.
3. Run the `jPScreenshotTest` from `JPScreenshotTest` in Android Studio to check that it **succeeds**.

   ⚠️ **Note:** _When running multiple times you may get fails with Hilt or sometimes wiremock. We are starting an investigation into upgrading our wiremock library (we are really, really, really, far behind). Hopefully that will fix the wiremock issues._

   <details>
   <summary>Emulator Setup Guide</summary>

     Ideally we run the tests on an emulator created by fastlane. See the notes in the issue for how this can be done:

     > Follow the guide in `SCREENSHOT_DEVICE_SETUP.md` to create a device like the ones we're using for screenshots automation (this won't be super easy as the guide is a bit outdated and if you're on an M1 Mac you have to change the generated .ini files manually to set the arm64 architecture or else the emultators won't start.
   </details>

4. Close emulator & wipe data
5. Switch to `wordpressJalapenoDebug` build flavor.
6. Run the `wPScreenshotTest` from `WPScreenshotTest` to check that it **succeeds**, meaning we're regression free.

## Regression Notes
1. Potential unintended areas of impact
WP Screenshot tests no longer work

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Ran WPScreenshot tests and validated they function as expected

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/JPScreenshotTest.java [hunk 1]
```diff
@@ -51,8 +63,39 @@ public class JPScreenshotTest extends BaseTest {
     @ClassRule
     public static final WPLocaleTestRule LOCALE_TEST_RULE = new WPLocaleTestRule();
 
+    private static final String JETPACK_SCREENSHOT_SITE_URL = "yourjetpack.blog";
+
     private DemoModeEnabler mDemoModeEnabler = new DemoModeEnabler();
 
+    public enum Screenshots {
```
Comment (ovitrif) [inline WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/JPScreenshotTest.java:70]: Nice enhancement here with the Screenshots enum 👏 👍

File: WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/JPScreenshotTest.java [hunk 2]
```diff
@@ -211,19 +466,15 @@ private boolean runningInTestLab() {
...
-    // In some cases there's a gray overlay on view pager screens when taking screenshots
-    // this function swipes left and then right as a workaround to clear it
-    // resourceID should be the ID of the viewPager
-    private void swipeToAvoidGrayOverlay(int resourceID) {
-        // Workaround to avoid gray overlay
-        swipeLeftOnViewPager(resourceID);
-        idleFor(1000);
-        swipeRightOnViewPager(resourceID);
-        idleFor(1000);
+    private void checkAndNavigateToSiteIfNotShowing() {
```
Comment (ovitrif) [inline WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/JPScreenshotTest.java:469]: This is another great enhancement that makes the work on the screenshots much easier now that we can order the tests in any way and easily disable the ones we're not working on while in development or debugging 👍 .

Thank you for the effort to make this possible @zwarm  🥇

File: WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/WPScreenshotTest.java [hunk 3]
```diff
@@ -80,6 +77,12 @@ public void wPScreenshotTest() {
     private void editBlogPost() {
         (new MySitesPage()).switchToSite("fourpawsdoggrooming.wordpress.com")
                            .goToPosts();
+
+        // There is a possibility of the edit post getting stuck with an `AppNotIdleException`
+        // On the UI it's shown by the flashing progress indicator at the bottom. This idle
+        // appears to wait long enough for the edit screen to show properly
+        idleFor(3000);
```
Comment (ovitrif) [inline WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/WPScreenshotTest.java:84]: Great finding, good thing I wasn't spending more time on it 😅 , I was on a totally different path that could have taken much more time.

**Review by ovitrif — APPROVED** [review]
Great work @zwarm! 🥇 

The code lgtm 🎉 , I appreciate the extra effort you put into making sure working on the JP screenshot tests gets easier.

Thank you for finding and applying the fix for the editor screenshot 🎉 .

In my runs the tests were green, and there's no regression with the WP screenshot tests 💚 🚀  :shipit:

**Review by AjeshRPai — APPROVED** [review]
@zwarm Everything looks good to me. Thanks for the improvements and the clean-up. 👍🏼 .

**Pull request merged**